exercises/practice/pov/tests/pov.rs

```diff
@@ -0,0 +1,428 @@
+                ],
+            )],
+        );
+        let from = "x".to_string();
+        let to = "sibling-1".to_string();
+        let result = input.path_to(&from, &to);
+        let expected = Some(vec![
+            "x".to_string(),
+            "parent".to_string(),
+            "sibling-1".to_string(),
+        ]);
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    #[ignore]
+    fn can_find_path_from_nodes_other_than_x() {
+        let input = Tree::with_children(
+            "parent".to_string(),
+            vec![
+                Tree::new("a".to_string()),
+                Tree::new("x".to_string()),
+                Tree::new("b".to_string()),
+                Tree::new("c".to_string()),
+            ],
+        );
+        let from = "a".to_string();
+        let to = "c".to_string();
+        let result = input.path_to(&from, &to);
+        let expected = Some(vec!["a".to_string(), "parent".to_string(), "c".to_string()]);
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    #[ignore]
+    fn errors_if_destination_does_not_exist() {
+        let input = Tree::with_children(
+            "parent".to_string(),
+            vec![
+                Tree::with_children(
+                    "x".to_string(),
+                    vec![
+                        Tree::new("kid-0".to_string()),
+                        Tree::new("kid-1".to_string()),
+                    ],
+                ),
+                Tree::new("sibling-0".to_string()),
+                Tree::new("sibling-1".to_string()),
+            ],
+        );
+        let from = "x".to_string();
+        let to = "nonexistent".to_string();
+        let result = input.path_to(&from, &to);
+        let expected: Option<Vec<String>> = None;
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    #[ignore]
+    fn errors_if_source_does_not_exist() {
+        let input = Tree::with_children(
+            "parent".to_string(),
+            vec![
+                Tree::with_children(
+                    "x".to_string(),
+                    vec![
+                        Tree::new("kid-0".to_string()),
+                        Tree::new("kid-1".to_string()),
+                    ],
+                ),
+                Tree::new("sibling-0".to_string()),
+                Tree::new("sibling-1".to_string()),
+            ],
+        );
+        let from = "nonexistent".to_string();
+        let to = "x".to_string();
+        let result = input.path_to(&from, &to);
+        let expected: Option<Vec<String>> = None;
+        assert_eq!(result, expected);
+    }
+}
+
+mod test_util {
+    use pov::*;
+
+    pub fn tree_option_eq(lhs: Option<Tree<String>>, rhs: Option<Tree<String>>) -> bool {
+        match (lhs, rhs) {
+            (None, None) => true,
+            (Some(l_inner), Some(r_inner)) => tree_eq(&l_inner, &r_inner),
+            _ => false,
+        }
+    }
+
+    pub fn tree_eq(lhs: &Tree<String>, rhs: &Tree<String>) -> bool {
```

Comment:
I would prefer the normal `Eq` traits be used instead of a separate function. Maybe you could implement a standalone function that only sorts the keys? Then you could get the tree from the user's solution, sort it, then compare it normally with the expected tree.